src/prefect/engine.py

```diff
@@ -410,10 +410,20 @@ async def retrieve_flow_then_begin_flow_run(
     - Updates the flow run version
     """
     flow_run = await client.read_flow_run(flow_run_id)
+
+    entrypoint = os.environ.get("PREFECT__FLOW_ENTRYPOINT")
+
     try:
-        flow = await load_flow_from_flow_run(flow_run, client=client)
+        flow = (
```

Comment:
Hmm, I think I'm missing the piece where we need this change to allow the engine to load flows from entrypoints as well as flow runs.